src/model.cpp

```diff
@@ -134,36 +131,30 @@ time_type model::run(time_type tfinal, time_type dt) {
     // events that must be delivered at the start of the next
     // integration period at the latest.
     auto exchange = [&] () {
-        PE("stepping", "communication");
-
-        PE("exchange");
+        PE(communication_exchange_gatherlocal);
         auto local_spikes = previous_spikes().gather();
-        auto global_spikes = communicator_.exchange(local_spikes);
         PL();
+        auto global_spikes = communicator_.exchange(local_spikes);
 
-        PE("spike output");
+        PE(communication_spikeio);
         local_export_callback_(local_spikes);
         global_export_callback_(global_spikes.values());
         PL();
 
-        PE("events","from-spikes");
+        PE(communication_walkspikes);
         communicator_.make_event_queues(global_spikes, pending_events_);
         PL();
 
-        PE("enqueue");
         const auto t0 = epoch_.tfinal;
         const auto t1 = std::min(tfinal, t0+t_interval);
         setup_events(t0, t1, epoch_.id);
-        PL(2);
-
-        PL(2);
     };
 
+    util::profiler_start(); // start the profilers just before time stepping begins
+
```

Comment:
Mentioned elsewhere, but tying the profiler active lifetime to `model::run()` means we can't use the profiler to track e.g. cell construction costs. As argued above, I think we should separate these concerns.